cmd/nerdctl/main.go

```diff
@@ -87,17 +88,41 @@ func xmain() error {
 // Config corresponds to nerdctl.toml .
 // See docs/config.md .
 type Config struct {
-	Debug            bool     `toml:"debug"`
-	DebugFull        bool     `toml:"debug_full"`
-	Address          string   `toml:"address"`
-	Namespace        string   `toml:"namespace"`
-	Snapshotter      string   `toml:"snapshotter"`
-	CNIPath          string   `toml:"cni_path"`
-	CNINetConfPath   string   `toml:"cni_netconfpath"`
-	DataRoot         string   `toml:"data_root"`
-	CgroupManager    string   `toml:"cgroup_manager"`
-	InsecureRegistry bool     `toml:"insecure_registry"`
-	HostsDir         []string `toml:"hosts_dir"`
+	Debug            bool         `toml:"debug"`
+	DebugFull        bool         `toml:"debug_full"`
+	Address          string       `toml:"address"`
+	Namespace        string       `toml:"namespace"`
+	Snapshotter      string       `toml:"snapshotter"`
+	CNIPath          string       `toml:"cni_path"`
+	CNINetConfPath   string       `toml:"cni_netconfpath"`
+	DataRoot         string       `toml:"data_root"`
+	CgroupManager    string       `toml:"cgroup_manager"`
+	InsecureRegistry bool         `toml:"insecure_registry"`
+	HostsDir         []string     `toml:"hosts_dir"`
+	RunConfig        *RunConfig   `toml:"default_run_config"`
```

Comment:
Shouldn't this be `map[string]...` where the key is a containerd namespace string such as "default"?

Reply:
I think both `map[string]...` and user-defined structs are legal, I find an example from https://pkg.go.dev/github.com/pelletier/go-toml@v1.9.5#section-readme and test it works.

Reply:
Ah, if I can reuse the compose structure, then here I can just use `map[namespace]ServiceConfig` to avoid defining multiple similar structures. I'll have some try, thanks!

Reply:
I think it is more appropriate to choose parsing to structs than to maps here, so that only fields defined in structs will be parsed, and unsupported fields will be ignored in parsing even if they are mistakenly written to the configuration file by the user.